Add LogAntrean model and migration for call logging

presidenwashil opened the pull request.

presidenwashil committed:
fix: Update status filter in antrianPerPintu scope to include 'TTV'

presidenwashil committed:
Merge pull request #177 from Rizky92/ubah-antrean-panggil

Ubah antrean panggil

presidenwashil committed:
Merge pull request #187 from Rizky92/ubah-antrean-panggil

feat: add display for current patient in examination

Rizky92 committed:
Disable translate

Signed-off-by: Muhammad Rizky Rizaldi <myquinn489@gmail.com>

presidenwashil committed:
fix: add sorting by doctor name in Pintu model's antrian query

presidenwashil committed:
fix: perubahan filter pemanggilan antrean

presidenwashil committed:
Merge pull request #196 from Rizky92/fix-kondisi-panggil

fix: perubahan filter pemanggilan antrean

presidenwashil committed:
Add LogAntrean model and migration for call logging

Introduces the LogAntrean model and migration to log called queue entries in the mysql_smc database. Updates AntreanDiPanggil to use LogAntrean for tracking the current called queue and refactors related queries and view bindings to use new relationships.

presidenwashil commented:
<html>
<body>
<!--StartFragment--><html><head></head><body>
<h3>🔍 Analisa Masalah</h3>
<p>Pada proses pemanggilan pasien saat ini, terdapat anomali ketika <strong>lebih dari satu dokter</strong> menggunakan <strong>pintu panggil yang sama</strong>.</p>
<p>Alur sistem berjalan seperti berikut:</p>
<ol>
<li>
<p>Pasien A no.1 dipanggil ⇒ status <code inline="">'1'</code> → setelah terpanggil diubah menjadi <code inline="">'0'</code> (sedang diperiksa).</p>
</li>
<li>
<p>Pasien A no.1 selesai ⇒ data dihapus dari DB, kemudian pasien A no.2 dipanggil ⇒ insert dengan status <code inline="">'1'</code>, lalu berubah lagi menjadi <code inline="">'0'</code> saat diperiksa.</p>
</li>
<li>
<p><strong>Masalah muncul</strong> saat masuk ke dokter B:</p>
<ul>
<li>
<p>Pasien B no.1 dipanggil ⇒ status <code inline="">'1'</code> → setelah status berubah <code inline="">'0'</code></p>
</li>
<li>
<p>Sistem <strong>salah menampilkan</strong> pasien yang sedang diperiksa ⇒ yang tampil justru <strong>pasien A no.2</strong></p>
</li>
<li>
<p>Hal ini terjadi karena query selalu mengambil pasien <code inline="">'status=0'</code> dari dokter dengan <strong>jadwal pertama</strong> di pintu tersebut, bukan dokter yang <strong>baru saja memanggil pasien</strong></p>
</li>
</ul>
</li>
</ol>
<p>⚠️ Artinya: Sistem kehilangan <em>context</em> dokter yang terakhir aktif di pintu tersebut, sehingga pasien dari dokter sebelumnya masih dianggap “sedang diperiksa”.</p>
<hr>
<h3>Penyebab Utama</h3>
<ul>
<li>
<p>Ketika di satu pintu terdapat <strong>lebih dari dua jadwal dokter</strong>, sistem tidak mengetahui dokter mana yang sebenarnya <strong>sedang aktif</strong> memanggil pasien.</p>
</li>
<li>
<p>Urutan jadwal tidak bisa dijadikan acuan karena contoh kasus berikut gagal:</p>
<pre><code>08.00 - dr. A
09.00 - dr. B
10.00 - dr. A
</code></pre>
<p>Ketika masuk ke jadwal dr. B, sistem tetap menampilkan pasien terakhir dr. A (karena jadwal dr. A paling akhir).</p>
</li>
</ul>
<hr>
<h3>Evaluasi Solusi</h3>

Solusi | Deskripsi | Kelebihan | Kekurangan
-- | -- | -- | --
A | Mengandalkan rentang waktu jadwal dokter | Selalu sesuai jadwal aktif saat ini | Tidak cocok bila jam eksisting di lapangan tidak sesuai jadwal sistem
B | Menambahkan log table untuk mencatat pemanggilan pasien | Sistem bisa merujuk pemanggilan terakhir per pintu (real-time) | Perlu perubahan proses pemanggilan agar insert log dilakukan setiap kali pasien dipanggil


<hr>
<h3>Rekomendasi</h3>
<p>✅ <strong>Solusi B lebih ideal</strong> karena:</p>
<ol>
<li>
<p>Tidak tergantung pada jadwal dokter yang bisa berubah atau tidak tepat waktu.</p>
</li>
<li>
<p>Menjaga <em>context</em> pemanggilan: pintu → dokter → pasien terakhir.</p>
</li>
<li>
<p>Query “sedang diperiksa” cukup mengacu pada <strong>log terakhir</strong> untuk pintu tersebut.</p>
</li>
</ol>
<p>Implementasi tambahan:</p>
<ul>
<li>
<p>Saat pasien dipanggil ⇒ insert record ke log table</p>
</li>
<li>
<p>Field minimal: <code inline="">kd_pintu</code>, <code inline="">kd_dokter</code>, <code inline="">no_rawat/no_reg</code>, <code inline="">timestamp</code></p>
</li>
<li>
<p>Saat display menampilkan <strong>sedang diperiksa</strong> ⇒ ambil record terbaru dari log table berdasarkan <code inline="">kd_pintu</code>.</p>
</li>
</ul>
<hr>
<h3>Kesimpulan</h3>
<p>Dengan log table, sistem akan selalu mengetahui <strong>dokter aktif terbaru</strong> pada pintu panggil, sehingga tidak terjadi lagi kesalahan penampilan pasien yang sedang diperiksa meskipun ada beberapa jadwal dokter dalam satu pintu dengan catatan kita perlu merubah proses pemanggilan di sistem pemanggilannya</p>
<!--EndFragment-->
</body>
</html>

presidenwashil committed:
Revert "Add LogAntrean model and migration for call logging"

This reverts commit ff637653e4144778b7516c6950e2a42795570c52.

presidenwashil committed:
wip

presidenwashil committed:
wip

Rizky92 reviewed (approved):
Mungkin perlu job buat cleaning antrian dengan status = 0? Atau ada ide lain?

presidenwashil committed:
wip

Rizky92 commented:
> Mungkin perlu job buat cleaning antrian dengan status = 0? Atau ada ide lain?

Jangan lupa ini ya @presidenwashil, nanti kalau sudah masuk #189 dan #191 kita coba test.

presidenwashil committed:
wip

presidenwashil committed:
Add daily cleanup task to console kernel

Introduces a scheduled job that deletes records with status '0' from the 'antripintu_smc' table in the 'mysql_sik' connection every day at Asia/Singapore timezone.

presidenwashil committed:
Merge branch 'live' into add-log-antrean

Rizky92 commented on app/Livewire/Pages/Antrean/AntreanDiPanggil.php:
```diff
@@ -60,27 +46,75 @@ public function call(): void
+            if (method_exists($this, 'dispatch')) {
+                // Livewire v3
+                $this->dispatch('play-voice',
+                    no_reg: $antrean->no_reg,
+                    nm_pasien: $antrean->nm_pasien,
+                    nm_pintu: $antrean->nm_pintu
```
Target minimum PHP masih ^7.4, named argument baru ada di PHP 8.

presidenwashil replied:
fixed
https://github.com/Rizky92/smc-internal-app/commit/c8d0d7d28e65b92a80a1c88427f5f940324ed4c4
